### PR TITLE
Basic access control

### PR DESCRIPTION
**Pull request by mandarjog**

### CODE REVIEW

File: _docs/tasks/basic-access-control.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  istioctl replace -f route-rule-reviews-v2-v3.yaml
+  ```
+* Ensure that you can use [istioctl mixer](/docs/reference/commands/istioctl/istioctl_mixer.html#synopsis) by setting up port forwarding if needed.
+
+## Access control using `denials` 
+
+Using Istio you can control access to a service based on any attributes that are available within Mixer.
+This simple form of access control is based on conditionally denying requests using Mixer selectors.
+
+Consider the [bookinfo](/docs/samples/bookinfo.html) sample application where `ratings` service is accessed by multiple versions
```
Comment (geeknoid) [inline _docs/tasks/basic-access-control.md]: bookinfo -> BookInfo
URL -> ({{home}}/docs/samples/bookinfo.html)
`ratings` -> the `ratings`

File: _docs/tasks/basic-access-control.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+```yaml
+rules:
+- selector: source.labels["app"]=="reviews" && source.labels["version"] == "v3"  
+  aspects:
+  - kind: denials
+```
+This rule uses the `denials` aspect to deny requests coming from version `v3` of the reviews service.
+
+## Access control using `whitelists` 
+Istio also supports attribute based white and black lists.
```
Comment (geeknoid) [inline _docs/tasks/basic-access-control.md]: Blank line.
attribute based -> attribute-based
white list -> whitelist (here and elsewhere)
black list -> blacklist (here and elsewhere)

File: _docs/tasks/basic-access-control.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+2 Enable white list checking by using the `lists` aspect:
+
+```yaml
+rules:
+  aspects:
+  - kind: lists
+    adapter: versionList
+    params:
+      blackList: false
```
Comment (geeknoid) [inline _docs/tasks/basic-access-control.md]: We should rename this symbol so it doesn't have an uppercase L. 'blacklist' is a single word.

File: _docs/tasks/basic-access-control.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      checkExpression: source.labels["version"] 
+``` 
+`checkExpression` is evaluated and checked against the list. This configuration will only allow
+access from versions `v1, v2`.
+
+
+## What's next
+* Learn more about [Mixer](/docs/concepts/policy-and-control/mixer.html) and [Mixer Config](/docs/concepts/policy-and-control/mixer-config.html).
```
Comment (geeknoid) [inline _docs/tasks/basic-access-control.md]: Blank line after header, blank line between each bullet.

Comment (geeknoid) [inline _docs/tasks/basic-access-control.md]: Add {{home}} in front or the URL here and following lines.

File: _docs/tasks/basic-access-control.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+```
+where deny-reviews.yml is 
+ 
+```yaml
+rules:
+- selector: source.labels["app"]=="reviews" && source.labels["version"] == "v3"  
+  aspects:
+  - kind: denials
+```
+This rule uses the `denials` aspect to deny requests coming from version `v3` of the reviews service.
```
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: do we want some "Understanding" subsection (or full section below) that explains the denials aspect?

Also, can we link somewhere to a reference guide on denials aspect config?

File: _docs/tasks/basic-access-control.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+```yaml
+rules:
+  aspects:
+  - kind: lists
+    adapter: versionList
+    params:
+      blackList: false
+      checkExpression: source.labels["version"] 
+``` 
+`checkExpression` is evaluated and checked against the list. This configuration will only allow
```
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: as above, can we link to reference on list checker config?  and maybe add some Understanding section?

File: _docs/tasks/basic-access-control.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+where deny-reviews.yml is 
+ 
+```yaml
+rules:
+- selector: source.labels["app"]=="reviews" && source.labels["version"] == "v3"  
+  aspects:
+  - kind: denials
+```
+This rule uses the `denials` aspect to deny requests coming from version `v3` of the reviews service.
+
```
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: aren't we missing a test/verification step that shows the impact of `denials` ?

**Comment by geeknoid** [comment]
Zack is still setting up the process to produce reference pages for the
aspect & adapter configs, so we don't have anything to link to yet.

File: _docs/tasks/basic-access-control.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  aspects:
+  - kind: lists
+    adapter: versionList
+    params:
+      blackList: false
+      checkExpression: source.labels["version"] 
+``` 
+`checkExpression` is evaluated and checked against the list. This configuration will only allow
+access from versions `v1, v2`.
+
```
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: as above, I feel like we are missing a step that shows access to v1 and v2 and shows blocked access to v3.

File: _docs/tasks/basic-access-control.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
+---
+title: Enabling Simple Access Control
+overview: This task shows how to use Istio to control access to a service.
+          
+order: 41
+
+layout: docs
+type: markdown
+---
+
```
Comment (mandarjog) [inline _docs/tasks/basic-access-control.md]: None of the other tasks or the template use `include`. Is this something new we are setting up?

File: _docs/tasks/basic-access-control.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+This task shows how to use Istio to control access to a service.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the
+  [Installation guide](/docs/tasks/installing-istio.html).
+
+* Deploy the [bookinfo](/docs/samples/bookinfo.html) sample application.
```
Comment (mandarjog) [inline _docs/tasks/basic-access-control.md]: /docs/samples/bookinfo.html is consistent with all other tasks here. do we need a change PR for all those ?

**Comment by geeknoid** [comment]
Sync your bits, you'll see many tasks are now using {% include home.html %}
and all site references use {{home}}.

Checkout the
https://istio.io/docs/reference/contribute/writing-a-new-topic.html page
for details.

File: _docs/tasks/basic-access-control.md (hunk 5 above)
Comment (mandarjog) [inline _docs/tasks/basic-access-control.md]: I have added links to where Zack told me they would live.

**Comment by ZackButcher** [comment]
@mandarjog FYI https://github.com/istio/istio.github.io/pull/118 has the new protos you want to reference.

File: _docs/tasks/basic-access-control.md (hunk 8 above)
Comment (mandarjog) [inline _docs/tasks/basic-access-control.md]: Added verification steps.

File: _docs/tasks/basic-access-control.md (hunk 6 above)
Comment (mandarjog) [inline _docs/tasks/basic-access-control.md]: Added some more explanation, and a pointer to the generated docs.

**Comment by geeknoid** [comment]
Whoops, sorry for commiting this.

I thought this was an update that only changed the spelling for blacklist, I didn't realize it was part of the bigger PR you've been working on.

**Comment by douglas-reid** [comment]
Not sure where to mention this, but the use of "`denials`" in the subject header of `Access Control using denials` looks really odd, as the font size of denials doesn't match.

**Comment by douglas-reid** [comment]
There is also weird formatting in the "Access Control using whitelist" section.

```
1 Add an adapter definition for the genericListChecker) adapter that lists versions v1, v2:
```
- no '.' after numbers
- ')' where there shouldn't be one.

File: _docs/tasks/basic-access-control.md (hunk 8 above)
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: I don't see these in the docs?  Did they get lost, or have I missed something?

**Pull request merged**